pyartcd/pyartcd/jenkins.py

```diff
@@ -0,0 +1,67 @@
+import base64
+import logging
+import os
+from urllib.parse import urlencode
+
+from pyartcd import exectools
+
+logger = logging.getLogger(__name__)
+
+
+async def trigger_jenkins_job(job_path: str, params=None):
+    """
+    Trigger a job using remote API calls.
+    :param job_path: relative path to the job, starting from <buildvm hostname>:<jenkins port>
+    :param params: optional dict containing job parameters
+    """
+
+    service_account = os.getenv('JENKINS_SERVICE_ACCOUNT')
+    token = os.getenv('JENKINS_SERVICE_ACCOUNT_TOKEN')
+
+    if not service_account or not token:
+        logger.warning(
+            'Env vars JENKINS_SERVICE_ACCOUNT and JENKINS_SERVICE_ACCOUNT_TOKEN '
+            'must be defined in order to trigger jobs using remote API')
+        raise RuntimeError
+
+    # We need to use cURL --resolve option to pin the request to local host
+    # Doing so in pure Python requires a lot of hacking (see for example http://bit.ly/3ICezXw)
+    auth_header = base64.b64encode(f'{service_account}:{token}'.encode()).decode()
+    logger.info(f'auth_header: {auth_header}')
+
+    cmd = f'curl -X POST -H "Authorization: Basic {auth_header}" ' \
+          f'--resolve buildvm.hosts.prod.psi.bos.redhat.com:8443:127.0.0.1 ' \
+          f'https://buildvm.hosts.prod.psi.bos.redhat.com:8443/{job_path}'
+
+    # If the job to be triggered is parametrized, use 'buildWithParameters'
+    # and send form data with -d option
+    if params:
+        cmd += f'/buildWithParameters -d {urlencode(params)}'
+
+    # Otherwise, just call the 'build' endpoint
+    else:
+        cmd += '/build'
+
+    logger.info('Triggering remote job %s', job_path)
+    await exectools.cmd_assert_async(cmd)
+
+
+async def trigger_ocp4(build_version: str):
+    await trigger_jenkins_job(
+        job_path='job/triggered-builds/job/ocp4',
```

Comment:
Is this a new job or I missed something?

Reply:
Yeah, I explained this in the PR description:

> Since the aforementioned jobs are part of a multibranch pipeline, they cannot be configured to receive remote triggers. As a workaround, a new folder named triggered-jobs has been created in Jenkins; these jobs are acting like proxies between the caller ocp4-scan Python job, and the called jobs.